elastic/apm-server
# test backoff time across different runs

The idea behind this test is that, given there's
randomness in the jitter, across N requests
and a large enough max time, there should be an
appreciable variance between total time required
to execute N requests because of the jitter.

Whether or not that is a good idea, or if someone
has a better way to test for randomness, is up to
the reader.

The tests were parallelized so that multiple
backoff cycles could be tested at the same time
without hitting the 10min golang test timeout, and
also just because who wants to wait that long.

For reference, the current test setup has
completed full runs in:
- [5m35.752209718s 6m37.949983435s 6m52.242817354s]
- [5m2.881926583s 5m27.0031564s 5m49.786217507s]



## Motivation/summary

cf. https://github.com/elastic/apm-server/issues/4138

## Checklist




For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these changes

You can add log statements to the test to verify the different request timings, or run the test.

## Related issues
